main/popup/popup.js

```diff
@@ -3,41 +3,34 @@ const inputPassword = document.getElementById("input-password");
 const inputDate = document.getElementById("input-date");
 const submitButton = document.getElementById("popup-submit");
 
+const users = "registeredUsers" in localStorage ? JSON.parse(localStorage.getItem("registeredUsers")) : [];
 
-if(localStorage.getItem("registeredUsers") === null) {
-    localStorage.setItem("registeredUsers", JSON.stringify([])); 
-}
-
-let users = JSON.parse(localStorage.getItem("registeredUsers"));
 
 submitButton.addEventListener("click",function() {
-    return new Promise(function(resolve,reject) {
-        const usernameRegexp = /^\w+$/;
-        const passwordRegexp = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
-        setTimeout(function() {
-            if(passwordRegexp.test(inputPassword.value) && usernameRegexp.test(inputUserName.value) ) {
-                resolve(users.push({
-                    username: inputUserName.value,
-                    password: inputPassword.value,
-                    date: inputDate.value,
-                    registered: true,
-                    tickets: {
-                        interstellar: [],
-                        dunePartTwo: [],
-                        captainMiller: [],
-                        theBeekeeper: [],
-                        suriyasKanguva: []
-                    }
-                }))
-            } else {
-                reject("invalid Registration")
-            }
-        },1000)
-    })
 
-    .then(function(result) {
-        localStorage.setItem("registeredUsers",JSON.stringify(users))
-        alert("Registration done succesfully")
-    }).catch(error => alert(error))
-})
+    const usernameRegexp = /^\w+$/;
+    const passwordRegexp = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;    
+
+        if(passwordRegexp.test(inputPassword.value) && usernameRegexp.test(inputUserName.value)) {
+            alert("registration done succesfully")
+            users.push({
+                username: inputUserName.value,
+                password: inputPassword.value,
+                date: inputDate.value,
+                isRegistered: true,
+                theme: "ligth",
+                tickets: {
+
+                }
+            })
+
+            localStorage.setItem("registeredUsers",JSON.stringify(users))
+            document.location.href = "../login/login.html"
+
+        }
+        
+        else {
+            alert("invalid Registration");
```

Comment:
you should handle cases of validation.

1. Your password is too short
2. Your password should contain uppercase, number and symbol
3. Maximum password length is 16 

and etc.